lib/inc/internal/util/agent.hpp

```diff
@@ -0,0 +1,22 @@
+/**
+ * @file
+ * Declares methods for Facter's integration with the puppet-agent package.
+ */
+#pragma once
+
+#include <leatherman/execution/execution.hpp>
+#include <string>
+
+namespace facter { namespace util { namespace agent {
+    inline std::string which(std::string exe) {
+#ifdef FACTER_PATH
+        std::string fixed = leatherman::execution::which(exe, {FACTER_PATH});
+        if (fixed.empty()) {
```

Comment:
Yeah, that's much better. This was just copied from what was already being copypasta'd around the codebase